### PR TITLE
Cranelift: remove logging of vcode when the log level isn't debug or more

### PR DESCRIPTION
**Pull request by bnjbvr**
This logging step may be quite expensive, since rendering the VCode into strings has never been
optimized at all and is using pretty naive string concatenation. Removing it is a clear compile times win for a large wasm module I'm looking at. These are the results with parallel compilation enabled, so there could be an overall speedup due to less malloc lock contention too.

- x64 (AMD Ryzen 9 3950X): from 5.2 seconds to 1.8 seconds, i.e. 65% speedup
- aarch64 (Mac M1): from 17.6 seconds to 6.4 seconds, i.e. 63% speedup

### CODE REVIEW

**Review by alexcrichton — APPROVED** [review]

File: cranelift/codegen/src/machinst/compile.rs [hunk 1]
```diff
@@ -29,10 +29,15 @@ where
         lower.lower(b)?
     };
 
-    debug!(
-        "vcode from lowering: \n{}",
-        vcode.show_rru(Some(b.reg_universe()))
-    );
+    // Creating the vcode string representation may be costly for large functions, so don't do it
+    // if the Debug level hasn't been statically (through features) or dyanmically (through
```
Comment (bjorn3) [inline cranelift/codegen/src/machinst/compile.rs:33]:
```suggestion
    // if the Debug level hasn't been statically (through features) or dynamically (through
```

**Comment by cfallin** [comment]
This is fantastic, thanks! I'm curious: what logging crate did you have in your environment (or was this just in wasmtime)? There was a related issue we found recently that was fixed in tokio-rs/tracing#1247; though if a similar issue occurs with e.g. `env_logger` then that's good to know too, and we should audit regalloc.rs for unnecessary expression evals in tracing!

**Comment by alexcrichton** [comment]
For posterity we had [more discussion on Zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift/topic/logging) about this and I believe that this is actually not necessary and I shouldn't have approved so quickly. I was under the impression that the arguments to logging functions were always evaluated but that is not the case. Further investigation shows that @bnjbvr's local logger may be behaving badly which was causing the slowdown here.

**Comment by bnjbvr** [comment]
Yes, the `log::max_level` was set to the highest possible value by a dependency's default ctor. Sorry for the churn, will post a revert soonish.

**Pull request merged**